Solutions/Azure Cloud NGFW by Palo Alto Networks/Analytic Rules/CloudNGFW-UnusualThreatSignatures.yaml

```diff
@@ -0,0 +1,55 @@
+id: 89a86f70-615f-4a79-9621-6f68c50f365f
+name: CloudNGFW By Palo Alto Networks - Threat signatures from Unusual IP addresses 
+description: |
+  'Identifies Palo Alto Threat signatures from unusual IP addresses which are not historically seen. 
+  This detection is also leveraged and required for MDE and PAN Fusion scenario
+  https://docs.microsoft.com/Azure/sentinel/fusion-scenario-reference#network-request-to-tor-anonymization-service-followed-by-anomalous-traffic-flagged-by-palo-alto-networks-firewall'
+severity: Medium
+status: Available
+requiredDataConnectors:
+  - connectorId: CloudNgfwByPAN
+    dataTypes:
+      - fluentbit_CL
+queryFrequency: 1h
+queryPeriod: 7d
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - Discovery
+  - Exfiltration
+  - CommandAndControl
+relevantTechniques:
+  - T1046
+  - T1030
+  - T1071.001
+tags:
+  - Fusion
+query: |
+  let starttime = 7d;
+  let endtime = 1d;
+  let timeframe = 1h;
+  let HistThreshold = 25; 
+  let CurrThreshold = 10; 
+  let HistoricalThreats = fluentbit_CL
+  | where ident_s == "THREAT"
+  | where isnotempty(parse_json(Message).src_ip)

```

Comment:
Hi @davidokeyode ,the ColumnName 'parse_json(Message).src_ip' is invalid. The value must start with a letter or underscore, and contain only alphanumeric English characters., but found with message "Invalid data model. Thanks.

Reply:
@v-sabiraj - This section seems valid to me. If you have a look at the 

KQL allows the use of the **_parse_json()_** function to extract data from JSON-formatted fields. The naming convention in KQL allows for names that include periods as part of the field path notation. Therefore, the syntax 'parse_json(Message).src_ip' is generally acceptable in KQL.

**Sample log data**: https://github.com/davidokeyode/Azure-Sentinel/blob/palo-alto-networks-azure-cloudngfw/Sample%20Data/Custom/Azure%20Cloud%20NGFW%20by%20Palo%20Alto%20Networks/threat_logs.json